src/gewekediag.jl

```diff
@@ -22,9 +22,12 @@ function gewekediag(x::AbstractVector{<:Real}; first::Real=0.1, last::Real=0.5,
     n = length(x)
     x1 = x[1:round(Int, first * n)]
     x2 = x[round(Int, n - last * n + 1):n]
-    z =
-        (Statistics.mean(x1) - Statistics.mean(x2)) /
-        hypot(mcse(x1; kwargs...), mcse(x2; kwargs...))
+    T = float(eltype(x))
+    s = hypot(
+        Base.first(mcse(Statistics.mean, reshape(x1, :, 1, 1); split_chains=1, kwargs...)),
+        Base.first(mcse(Statistics.mean, reshape(x2, :, 1, 1); split_chains=1, kwargs...)),
```

Comment:
```suggestion
        Base.first(mcse(Statistics.mean, reshape(x1, Val(3)); split_chains=1, kwargs...)),
        Base.first(mcse(Statistics.mean, reshape(x2, Val(3)); split_chains=1, kwargs...)),
```
(the `Base` was a bit surprising before I noticed the variable named `first`)

Reply:
Is the `reshape(A, ::Val)` method documented anywhere? I hadn't seen it before and couldn't find it in the Julia docs. I also found this old discussion where the consensus seemed to be to replace it: https://github.com/JuliaLang/julia/issues/27474

Reply:
I used it several times (was quite convenient sometimes) but I didn't know that there was some discussion about removing it. Maybe let's keep `reshape(x1, :, 1, 1)` in that case.